src/Packetery/Core/Entity/CustomsDeclaration.php

```diff
@@ -0,0 +1,336 @@
+<?php
+/**
+ * Class CustomsDeclaration
+ *
+ * @package Packetery
+ */
+
+declare(strict_types=1);
+
+namespace Packetery\Core\Entity;
+
+/**
+ * Class CustomsDeclaration
+ */
+class CustomsDeclaration {
+
+	/**
+	 * Unique identifier.
+	 *
+	 * @var string
+	 */
+	private $id;
+
+	/**
+	 * Order.
+	 *
+	 * @var Order
+	 */
+	private $order;
+
+	/**
+	 * EAD.
+	 *
+	 * @var string
+	 */
+	private $ead;
+
+	/**
+	 * Delivery cost.
+	 *
+	 * @var float
+	 */
+	private $deliveryCost;
+
+	/**
+	 * Invoice number.
+	 *
+	 * @var string
+	 */
+	private $invoiceNumber;
+
+	/**
+	 * Invoice issue date.
+	 *
+	 * @var \DateTimeImmutable
+	 */
+	private $invoiceIssueDate;
+
+	/**
+	 * Invoice.
+	 *
+	 * @var callable|null
+	 */
+	private $invoiceFile = null;
+
+	/**
+	 * Invoice file ID.
+	 *
+	 * @var string|null
+	 */
+	private $invoiceFileId = null;
+
+	/**
+	 * MRN.
+	 *
+	 * @var string|null
+	 */
+	private $mrn = null;
+
+	/**
+	 * Ead file.
+	 *
+	 * @var callable|null
+	 */
+	private $eadFile = null;
+
+	/**
+	 * EAD file ID.
+	 *
+	 * @var string|null
+	 */
+	private $eadFileId = null;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string|null        $id               ID.
+	 * @param Order              $order            Order.
+	 * @param string             $ead              Ead.
+	 * @param float              $deliveryCost     Delivery cost.
+	 * @param string             $invoiceNumber    Invoice number.
+	 * @param \DateTimeImmutable $invoiceIssueDate Invoice issue date.
+	 */
+	public function __construct(
+		?string $id,
+		Order $order,
+		string $ead,
+		float $deliveryCost,
+		string $invoiceNumber,
+		\DateTimeImmutable $invoiceIssueDate
+	) {
+		$this->id               = $id;
+		$this->order            = $order;
+		$this->ead              = $ead;
+		$this->deliveryCost     = $deliveryCost;
+		$this->invoiceNumber    = $invoiceNumber;
+		$this->invoiceIssueDate = $invoiceIssueDate;
+	}
+
+	/**
+	 * Gets invoice file ID.
+	 *
+	 * @return string|null
+	 */
+	public function getInvoiceFileId(): ?string {
+		return $this->invoiceFileId;
+	}
+
+	/**
+	 * Sets invoice file ID.
+	 *
+	 * @param string|null $invoiceFileId Invoice file ID.
+	 * @return void
+	 */
+	public function setInvoiceFileId( ?string $invoiceFileId ): void {
+		$this->invoiceFileId = $invoiceFileId;
+	}
+
+	/**
+	 * Gets EAD file ID.
+	 *
+	 * @return string|null
+	 */
+	public function getEadFileId(): ?string {
+		return $this->eadFileId;
+	}
+
+	/**
+	 * Sets EAD file ID.
+	 *
+	 * @param string|null $eadFileId EAD file ID.
+	 * @return void
+	 */
+	public function setEadFileId( ?string $eadFileId ): void {
+		$this->eadFileId = $eadFileId;
+	}
+
+	/**
+	 * Sets ID.
+	 *
+	 * @param string $id ID.
+	 * @return void
+	 */
+	public function setId( string $id ): void {
```

Comment:
Je nějaká logika v řazení těch metod? Udělejme to buď podle property nebo funkce.

Reply:
řadím podle názvu metody

Reply:
To není pravda. Navíc to umí IDE, takže bych volil řazení jiné.

Reply:
Změny jsem ještě nepushnul. Moje IDE umí řadit jen podle názvu metody.